### ISSUE TITLE
Is it possible to ask the bot to use xelatex/lualatex?

### ISSUE THREAD

**Issue by ice1000** [vdrhtc/InLaTeXbot#20]
Sometimes we want to typo CJK or emojis

**Comment by vdrhtc** [comment]
No, sorry, right now there is no such option.

I think I could do that in principle (just replaced pdflatex with xelatex):
![image](https://user-images.githubusercontent.com/3819012/141985448-97ad6fdc-259c-4d62-a5e2-90cf2213f5f9.png)
But that will require more work from me and debug, and I don't know when I will have free time. By the way, I couldn't find an easy way to include emoji, maybe you can recommend anything?

**Comment by ice1000** [comment]
I don't know. I was just googling and they said you need luatex😭

**Comment by ice1000** [comment]
Thanks for creating the bot! It's lovely

**Comment by Frederisk** [comment]
Maybe this will help? I modified and implemented [InXeLaTeXbot](https://github.com/Frederisk/InXeLaTeXbot) on the basis of this repository, and now it is also available on Telegram: [@InXeLaTeXbot](https://t.me/InXeLaTeXbot). It may be a bit slow due to server performance though, you can also clone the repository and build it on your own device.

For CJK text, @InXeLaTeXbot has built-in fonts such as [Source Han](https://github.com/search?q=org%3Aadobe-fonts+source-han), you only need to add text like this in the preamble to use:

```latex
\documentclass[fontset=none, UTF8]{ctexart}
\usepackage{xeCJK} % CJK font
  \setCJKmainfont{Source Han Serif} % or `Source Han Serif TC` for Traditional Chinese, etc
  \setCJKsansfont{Source Han Sans}
  \setCJKmonofont{Source Han Sans HW}
```

**Comment by vdrhtc** [comment]
Thank you very much for testing! Maybe you could run it for some time and if there are no issues, I then may replace pdflatex by xelatex? And if you have time, may be you cold test lualatex as well?

**Comment by amireza007** [comment]
> I've been using it for a few months, and everything seems to be working fine. In fact, I've been using it for quite some time, I've only recently sorted it out and made it public. I've recently been trying to write some scripts to make this bot easier to deploy, and some of them have been pulled into my fork.
> 
> As for your bot, I think it might also be a good idea to keep using `pdflatex` until switching is provided? In this case, users will now have `pdflatex` and `xelatex` to choose from (There are some rendering behavior differences between `pdflatex` and `xelatex`, such as `tikzpicture`). I'm also planning to add the toggle feature myself, but it's been slow due to being busy.
> 
> For `lualatex`, I have simply tested it before, and it seems to work as well. But since I'm mainly using `xelatex` instead of `lualatex`, the testing may not be very adequate.

Thay's a 

> Maybe this will help? I modified and implemented [InXeLaTeXbot](https://github.com/Frederisk/InXeLaTeXbot) on the basis of this repository, and now it is also available on Telegram: [@InXeLaTeXbot](https://t.me/InXeLaTeXbot). It may be a bit slow due to server performance though, you can also clone the repository and build it on your own device.
> 
> For CJK text, @InXeLaTeXbot has built-in fonts such as [Source Han](https://github.com/search?q=org%3Aadobe-fonts+source-han), you only need to add text like this in the preamble to use:
> 
> ```
> \documentclass[fontset=none, UTF8]{ctexart}
> \usepackage{xeCJK} % CJK font
>   \setCJKmainfont{Source Han Serif} % or `Source Han Serif TC` for Traditional Chinese, etc
>   \setCJKsansfont{Source Han Sans}
>   \setCJKmonofont{Source Han Sans HW}
> ```

Great work, man!
I was wondering how you did manage to find the built-in fonts of @InLaTeXbot, 'cause I am desperately in need of a Persian-supported font like Arial for using the XePersian package...

**Comment by vdrhtc** [comment]
@amireza007, I believe that @Frederisk should be able to install some Persian fonts on his machine so that they become available to XeLaTeX. But that is on his good will...

**Comment by Frederisk** [comment]
Hmm, yes, InXeLaTeXbot runs on my own device, so I can install the fonts I need myself.

**Comment by amireza007** [comment]
@Frederisk Yes yes, I thought there are some fonts(like the one you mentioned) included in the original bot, not the one you forked. My bad!
